### ISSUE TITLE
Status 400 response with request.ext.prebid.targeting: {includewinners: false, includebidderkeys: false}

### ISSUE THREAD

**Issue by butavicius** [prebid/prebid-server#3093]
Hello, 

### The bug

if I send a request to our prebid server with `request.ext.prebid.targeting: {includewinners: false, includebidderkeys: false}` payload, it fails with status 400 with seemingly unrelated error message:

`Invalid request: request.imp[0].ext.prebid.bidder contains unknown bidder: pubmaticS2S. Did you forget an alias in request.ext.prebid.aliases?`. "unknown" bidder could be adform too, or any other we've configured.

### Expected behavior

With `request.ext.prebid.targeting: {includewinners: false, includebidderkeys: false}` payload I expect to receive prebid server response with `response.seatbid[i].bid[j].ext.prebid.targeting` `undefined` or `{}`.

### Context

There is of course ```ext.prebid.aliases: { "pubmaticS2S": "pubmatic", "adformS2S": "adform", etc... }``` in my payload.

Requests with `request.ext.prebid.targeting: {includewinners: true, includebidderkeys: false}` work as expected. Also requests with `request.ext.prebid.targeting: {includewinners: false, includebidderkeys: true}` work as expected. It's only when I change them both to false it fails with this seemingly unrelated error.

Basically we don't want get any targeting from prebid server, as we use our custom prefix instead of `hb_`.

### Prebid server version:

```
"revision": "de6ed827885e8ab02efd8d94b7b16157bd4d9538",
"version": "0.234.0-3-gde6ed827"
```

<img width="855" alt="Screenshot 2023-09-11 at 17 40 02" src="https://github.com/prebid/prebid-server/assets/61953616/a16be69d-fe53-488d-a295-61b5786b85ae">

<img width="1115" alt="Screenshot 2023-09-11 at 17 47 59" src="https://github.com/prebid/prebid-server/assets/61953616/41aa1b68-9ecb-4f21-bb34-21ada09708e6">

I would appreciate any help regarding this, thanks.

**Comment by gargcreation1992** [comment]
@butavicius Could you please share the request json for which you are facing this issue?

**Comment by butavicius** [comment]
@gargcreation1992 sure, here is the full payload json:

```
{"imp":[{"ext":{"tid":"01b965d2-66d3-43f8-b4d6-ef75774f8d53","prebid":{"bidder":{"adformS2S":{"mid":859365,"adxDomain":"adx.adform.net"},"pubmaticS2S":{"publisherId":"156191","adSlot":"bannertag.com_300x600_S2S@300x600"},"smartadserverS2S":{"networkId":3527}},"adunitcode":"bannertag_com_300x600_universal_head"}},"id":"bannertag_com_300x600_universal_head","banner":{"topframe":1,"format":[{"w":300,"h":600},{"w":160,"h":600},{"w":300,"h":250},{"w":300,"h":300}]}}],"site":{"domain":"bannertag.com","publisher":{"domain":"bannertag.com","id":"1"},"page":"https://www.bannertag.com/universal_creative.html"},"regs":{"ext":{"gdpr":1}},"user":{"ext":{"consent":"CPx73UAPx73UAF7ABBENDWCsAP_AAH_AABaoJqtV_H__bW9r8X7_aft0eY1P9_j77uQxBhfJE-4F3LvW_JwXx2E5NF36tqoKmRoEu3ZBIUNlHJHUTVmwKokVryHsak2cpTNKJ6BkkFMRM2dYCF5vm5tjeQKY5_p_d3fx2D-t_dv839z3z81Xn3d5f--0-PCdU5-9Dfn9fRfb-9IP9_78v8v8_l_rk2_eT13_p8_r_D--f_87_XW-4JqAEmGhcQBdgQEhNoGEUCAEYVhARQKAAAASBogIAXBgU7IwCXWAiAECKAA4IAQgAoyABAAABAAhEAEgRAIAAIBAIAAQAIBAIACBgAFABYCAQAAgOgYphQAKBYQJEJEQpgQhQJBAS2VCCUFwgrhAEWWAFAIjYKABAAAIrAAEBYvAYAkBKhIIEuoNoAACABAKKUKxBJ6YABwSNlqDwQAA.YAAAAAAAAAAA","ConsentedProvidersSettings":{"consented_providers":"1~39.43.46.55.61.70.83.89.93.108.117.122.124.131.135.136.143.144.147.149.159.162.167.171.192.196.202.211.218.228.230.239.241.259.266.272.286.291.311.317.322.323.326.327.338.367.371.385.389.394.397.407.413.415.424.430.436.440.445.449.453.482.486.491.494.495.501.503.505.522.523.540.550.559.560.568.574.576.584.587.591.733.737.745.780.787.802.803.817.820.821.829.839.864.867.874.899.904.922.931.938.979.981.985.1003.1024.1027.1031.1033.1034.1040.1046.1051.1053.1067.1085.1092.1095.1097.1099.1107.1127.1135.1143.1149.1152.1162.1166.1186.1188.1201.1205.1211.1215.1226.1227.1230.1252.1268.1270.1276.1284.1286.1290.1301.1307.1312.1345.1356.1364.1365.1375.1403.1415.1416.1419.1440.1442.1449.1455.1456.1465.1495.1512.1516.1525.1540.1548.1555.1558.1564.1570.1577.1579.1583.1584.1591.1603.1616.1638.1651.1653.1665.1667.1677.1678.1682.1697.1699.1703.1712.1716.1721.1725.1732.1745.1750.1765.1769.1782.1786.1800.1808.1810.1825.1827.1832.1837.1838.1840.1842.1843.1845.1859.1866.1870.1878.1880.1889.1899.1917.1929.1942.1944.1962.1963.1964.1967.1968.1969.1978.2003.2007.2008.2027.2035.2039.2044.2046.2047.2052.2056.2064.2068.2070.2072.2074.2088.2090.2103.2107.2109.2115.2124.2130.2133.2137.2140.2145.2147.2150.2156.2166.2177.2183.2186.2202.2205.2216.2219.2220.2222.2225.2234.2253.2264.2279.2282.2292.2299.2305.2309.2312.2316.2322.2325.2328.2331.2334.2335.2336.2337.2343.2354.2357.2358.2359.2366.2370.2376.2377.2387.2392.2394.2400.2403.2405.2407.2411.2414.2416.2418.2425.2427.2440.2447.2459.2461.2462.2465.2468.2472.2477.2481.2484.2486.2488.2492.2493.2496.2497.2498.2499.2501.2510.2511.2517.2526.2527.2532.2534.2535.2542.2544.2552.2563.2564.2567.2568.2569.2571.2572.2575.2577.2583.2584.2595.2596.2601.2604.2605.2608.2609.2610.2612.2614.2621.2628.2629.2633.2634.2636.2642.2643.2645.2646.2647.2650.2651.2652.2656.2657.2658.2660.2661.2669.2670.2677.2681.2684.2686.2687.2690.2695.2698.2707.2713.2714.2729.2739.2767.2768.2770.2772.2784.2787.2791.2792.2798.2801.2805.2812.2813.2816.2817.2818.2821.2822.2827.2830.2831.2834.2836.2838.2839.2840.2844.2846.2847.2849.2850.2851.2852.2854.2856.2860.2862.2863.2865.2867.2869.2873.2874.2875.2876.2878.2880.2881.2882.2883.2884.2886.2887.2888.2889.2891.2893.2894.2895.2897.2898.2900.2901.2908.2909.2911.2912.2913.2914.2916.2917.2918.2919.2920.2922.2923.2924.2927.2929.2930.2931.2939.2940.2941.2942.2947.2949.2950.2956.2961.2962.2963.2964.2965.2966.2968.2970.2973.2974.2975.2979.2980.2981.2983.2985.2986.2987.2991.2993.2994.2995.2997.2999.3000.3002.3003.3005.3008.3009.3010.3012.3016.3017.3018.3019.3024.3025.3028.3034.3037.3038.3043.3045.3048.3052.3053.3055.3058.3059.3063.3065.3066.3068.3070.3072.3073.3074.3075.3076.3077.3078.3089.3090.3093.3094.3095.3097.3099.3100.3104.3106.3109.3112.3116.3117.3118.3119.3120.3124.3126.3127.3128.3130.3135.3136.3145.3149.3150.3151.3154.3155.3162.3163.3167.3172.3173.3180.3182.3183.3184.3185.3187.3188.3189.3190.3194.3196.3197.3209.3210.3211.3214.3215.3217.3219.3222.3223.3225.3226.3227.3228.3230.3231.3232.3234.3235.3236.3237.3238.3240.3241.3244.3245.3250.3251.3253.3257.3260.3268.3270.3272.3281.3288.3290.3292.3293.3295.3296.3300"},"eids":[{"source":"id5-sync.com","uids":[{"id":"ID5*TRUjsnjkK5GOmuI4UF-om9cpcoDZ0-bv2QPC6oNKjbhg5AdL2mfsX39svWFx33tLYOfYnu1i0nzq5R89ddf5g2Dowo2eRujlac_3wW-PTG1g6aGRMCYCu3tS4ikos3FrYOqtCePcVj1MrfAJKLKv1mDrvHwjMcTSMf4fSsFfzcxg7KSywLf5hzz20FZ_Q_DFYO3t3G0URmy-pO7IpzYzW2DzOwzxRSGOB4l3v4nhOgdg9Lf5Tr081bIJq1vEx0U6YPUh4gzDdRvZvExP5qC8MWD4aaDFeEtHW8jCpb5tLJVg-mQqY30MCdZZhUxM28QIYPzOOgtQWkfBmQ5WFCZUK2D9Rc9LAbr4EiPk1ubME5hg_pCrhtHhNDTQmF-u962VYQMkUKulg3cE-NBcFXqnCmEEm5Us3RmGcRsN8VJojg9hBUQ6V0LGc937w89hSY1IYQe4LlveBenfUAY_Jh28UmEJ5guU2II_2wghh19b3j5hC11Au4bUjZKrAxoBX9oYYQ8mCyQelbxohc0GTmk0LGEVHmPDhQtrWWi8XGMSbJNg5pWgJcwHONfJH7ZVCsPg","atype":1,"ext":{"linkType":2,"pba":"+iRoovR01QZutaxyiPlNKQ=="}}]},{"source":"criteo.com","uids":[{"id":"vv9mFV93eURSRE5rYnBJOXFwd3BjYmVTMGFNRnRpMzRjMkdwT1p6VTUydlVlTmdPNDlnU1pEV3FCNjc1UHI3cXVDTGJHZ2Y3VHJPbE5pN2k3ajYxbHo1N1pkY2hMTzdTYkFpWG8yZER3RXdBTlRnSG9TamNnbkRJOHMzWnhNZTkyWWJWckdPVFhxNUp3UE1Ld01teUljUHM1dEElM0QlM0Q","atype":1}]}]}},"device":{"w":741,"h":789,"dnt":0,"ua":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36","language":"en"},"ext":{"prebid":{"auctiontimestamp":1694499570278,"targeting":{"includewinners":false,"includebidderkeys":false},"aliases":{"pubmaticS2S":"pubmatic","openxS2S":"openx","sovrnS2S":"sovrn","adformS2S":"adform","adform1S2S":"adform","adform2S2S":"adform","pulsepointS2S":"pulsepoint","rtbhouseS2S":"rtbhouse","improvedigitalS2S":"improvedigital","rubiconS2S":"rubicon","appnexusS2S":"appnexus","emx_digitalS2S":"emx_digital","ixS2S":"ix","smartadserverS2S":"smartadserver"},"schains":[{"bidders":["adformS2S","pubmaticS2S","smartadserverS2S"],"schain":{"ver":"1.0","complete":1,"nodes":[{"asi":"setupad.com","sid":"363","hp":1}]}}],"channel":{"name":"pbjs","version":"v7.54.0"}}},"id":"85398f28-a53c-49d7-b27e-a8e1b5a047db","test":0,"source":{"tid":"85398f28-a53c-49d7-b27e-a8e1b5a047db","ext":{"schain":{"ver":"1.0","complete":1,"nodes":[{"asi":"setupad.com","sid":"363","hp":1}]}}},"cur":["GBP"],"tmax":150}
```

With this payload I got response 400:

```
Invalid request: request.imp[0].ext.prebid.bidder contains unknown bidder: adformS2S. Did you forget an alias in request.ext.prebid.aliases?
```

**Comment by gargcreation1992** [comment]
I checked this - running prebid-server locally with master branch. 
endpoint - http://localhost:8000/openrtb2/auction
First got an invalid alias reference for rubicon and ix so removed that from the request. Something like - `Invalid request: request.ext.prebid.aliases.rubiconS2S refers to disabled bidder: rubicon`

After that I got the valid response from the local server - `Invalid request: ext.prebid.targeting: At least one of includewinners or includebidderkeys must be enabled to enable targeting support`

Could you please re-verify this? Try running it in your local env against master branch.

**Comment by butavicius** [comment]
@gargcreation1992 I have re-verified it and confirmed that it works as you said.

So the unrelated error message is not an issue anymore in newer versions (I observed it in `v.0.234.0-3-gde6ed827`). 

But I was surprised to find out that a valid response is `"At least one of includewinners or includebidderkeys must be enabled to enable targeting support"`. The whole point of setting `{includewinners: false, includebidderkeys: false}` is to opt out of targeting support.

I see this has been put on the map in Prebid Server Prioritization: https://github.com/prebid/prebid-server/issues/2030

However, I think the label "enhancement" is strange, as implementing what this issue proposes seems more to me like a bug fix than a feature.

**Comment by gargcreation1992** [comment]
> But I was surprised to find out that a valid response is `"At least one of includewinners or includebidderkeys must be enabled to enable targeting support"`. The whole point of setting `{includewinners: false, includebidderkeys: false}` is to opt out of targeting support.
> 
> I see this has been put on the map in Prebid Server Prioritization: #2030
> 
> However, I think the label "enhancement" is strange, as implementing what this issue proposes seems more to me like a bug fix than a feature.

If you would like to exclude targeting then you can just not provide the targeting object in the json. That should solve your problem.

Hope this helps. Could we now please close this issue?

**Comment by butavicius** [comment]
@gargcreation1992 yes, thank you for looking at this.
